### ISSUE TITLE
Player disappears on Divinum V2

### ISSUE THREAD

**Issue by def-** [ddnet/ddnet#4285]
Join map, zoom out, player disappears.

@trml your area with the recent changes?

**Comment by fokkonaut** [comment]
This is the reason I have separated ENTTYPES for doors, etc in my mod (which @trml wasnt a fan of for ddnet, due to changes in processing order -> e.g. doors after characters)

It's not enough to only snap the player data before the entities, also the character entity itself has to be snapped before.
https://github.com/ddnet/ddnet/blob/82c7b20001b4e614b9992e3a44e149b908ce782a/src/game/server/gamecontext.cpp#L3625-L3637

Order, in which the server snaps and processes entities:
https://github.com/ddnet/ddnet/blob/82c7b20001b4e614b9992e3a44e149b908ce782a/src/game/server/gameworld.h#L23-L28

**Comment by trml** [comment]
Yes, although this problem already existed before with maps that had too many snap items when zooming out. Looked at this map, and it had 742 items before, just below the limit (1024), and it looks like it's all the pickups in one area that made it cross it.

And yes, the solution for keeping the characters is to snap them before others world items. To make things exactly like before the solution could be to increase the snap limit, but some other things like SnapFindItem should perhaps be optimized first.

**Comment by trml** [comment]
> This is the reason I have separated ENTTYPES for doors, etc in my mod (which @trml wasnt a fan of for ddnet, due to changes in processing order -> e.g. doors after characters)

Changing the order of ENTTYPES could risk changing physics, yes. Probably not for laser doors, and not sure about things like pickups and shotgun bullets, but for some things like lasers and projectiles/grenades (fired by the player) it definitely would.

A manual snap order seems like the best solution to me here, as it also lets you snap characters before those.
And if it turns out that those other things (the order of doors, shotgun, pickups, dragger, etc) don't change physics they could always be changed later.
